src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/v2/TaskListContent.tsx

```diff
@@ -19,39 +19,27 @@ export const TaskListContent = ({ project }: TaskListProps) => {
   const toast = useToast();
 
   const onDeleteTask = (taskId: string) => {
-    setTasksByStatus((prev) => {
-      const newTasksByStatus = { ...prev };
-      for (const status of statuses) {
-        const index = newTasksByStatus[status].findIndex((task) => task?.taskId === taskId);
-        if (index !== -1) {
-          newTasksByStatus[status].splice(index, 1);
-          break;
-        }
+    for (const status of statuses) {
+      const index = tasksByStatus[status].findIndex((task) => task?.taskId === taskId);
+      if (index !== -1) {
+        tasksByStatus[status].splice(index, 1);
```

Comment:
The change here turns a setState into a mutation of a state. Isn't this not allowed?

Reply:
yeah its prob not best practice i reverted it